jina/types/sets/document.py

```diff
@@ -30,6 +31,8 @@ def insert(self, index: int, doc: 'Document') -> None:
     def __setitem__(self, key, value: 'Document'):
         if isinstance(key, int):
             self._docs_proto[key].CopyFrom(value)
+        elif isinstance(key, UniqueId):
+            self._docs_map[str(key)].CopyFrom(value)
```

Comment:
I am not sure, if that is the right way to go. Perhaps we should also always convert into `UniqueId`, since it is the jina native `id`. Opinion?

Reply:
Thought about this a bit more. This can go away in the moment we remove the `__hash__` function from `UniqueId`, because afterwards `str` and `UniqueId` will get the same hash again.